aiida_common_workflows/workflows/dissociation.py

```diff
@@ -0,0 +1,216 @@
+    if len(value.sites) != 2:
+        return '`molecule`. only diatomic molecules are supported.'
+
+
+def validate_distances(value, _):
+    """Validate the `dinstances` input."""
+    if value and len(value) < 2:
+        return 'need at least 2 distances.'
+    if value:
+        for dist in value:
+            if dist < 0.0:
+                return 'distances must be positive.'
+
+
+def validate_distances_count(value, _):
+    """Validate the `dinstances_count` input."""
+    if value is not None and value < 2:
+        return 'need at least 2 distances.'
+
+
+def validate_distance_max(value, _):
+    """Validate the `distance_max` input."""
+    if value is not None and value <= 0:
+        return '`distance_max` must be bigger than zero.'
+
+
+def validate_distance_min(value, _):
+    """Validate the `distance_min` input."""
+    if value is not None and value < 0:
+        return '`distance_min` must be bigger than zero.'
+
+
+def validate_relax(value, _):
+    """Validate the `generator_inputs.relax_type` input."""
+    if value is not RelaxType.NONE:
+        return '`generator_inputs.relax_type`. Only `RelaxType.NONE` supported.'
+
+
+@calcfunction
+def set_distance(molecule: orm.StructureData, distance: orm.Float) -> orm.StructureData:
+    """
+    Move the second site of the molecule to meet a target distance
+    between the two sites of the molecule.
+    """
+    import numpy as np
+    vector_diff = np.array(molecule.sites[1].position) - np.array(molecule.sites[0].position)
+    versor_diff = vector_diff / np.linalg.norm(vector_diff)
+    molecule_new = molecule.clone()
+    new_position = np.array(molecule.sites[0].position) + distance.value * versor_diff
+    molecule_new.attributes['sites'][1]['position'] = new_position
+    return molecule_new
+
+
+class DissociationCurveWorkChain(WorkChain):
+    """Workflow to compute the dissociation curve of for a given diatomic molecule."""
+
+    @classmethod
+    def define(cls, spec):
+        # yapf: disable
+        super().define(spec)
+        spec.input('molecule', valid_type=orm.StructureData, validator=validate_molecule, help='The input molecule.')
+        spec.input('distances', valid_type=orm.List, required=False, validator=validate_distances,
+            help='The list of dinstances in Ang at which the total energy of the molecule should be computed.')
+        spec.input('distances_count', valid_type=orm.Int, default=lambda: orm.Int(20),
+            validator=validate_distances_count,
+            help='The number of points to compute in the dissociation curve.')
+        spec.input('distance_min', valid_type=orm.Float, default=lambda: orm.Float(0.5),
+            validator=validate_distance_min,
+            help='The minimum tested dinstance in Ang.')
+        spec.input('distance_max', valid_type=orm.Float, default=lambda: orm.Float(3),
+            validator=validate_distance_max,
+            help='The maximum tested distance in Ang.')
```

Comment:
Wondering if we should specify the default here. If someone specifies an explicit list of `distances` then the defaults will still be added as nodes. Notice now that the same is happening for the `EosWorkChain` so maybe ok to leave for now but maybe open a low priority issue to maybe find a better solution. Downside of not having the defaults here is that they cannot be introspected through the spec explicitly